qiskit/synthesis/clifford/clifford_decompose_layers.py

```diff
@@ -108,11 +110,15 @@ def synth_clifford_layers(
     """
 
     num_qubits = cliff.num_qubits
+    if cz_synth_func.__name__ == synth_cz_depth_line_mr.__name__:
```

Comment:
I am wondering if we can avoid hardcoding specific function names into the algorithm. I am thinking along the lines of adding an argument like `cz_reverse_synth_func`, and making the choice whether to follow the "clifford reversal" flow based on which of the two function `cz_synth_func` and `cz_reverse_synth_func` is not None.

Reply:
OK, I'll add a parameter called `reverse_qubits`